### PR TITLE
Sandbox: Fix dynamic loaded chunks not processed correcly inside the sandbox

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Fixes an issue when a plugin running inside the sandbox loads dynamic chunks. Sometimes those chunks get incorrect source map replaced and their native apis were not patched to make them compatible with the sandbox

**Why do we need this feature?**

To make plugins using dynamic chunks compatible with the sandbox

**Who is this feature for?**

plugins running inside the sandbox

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:

- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/cdn/utils.ts [hunk 1]
```diff
@@ -7,19 +7,23 @@ export function transformPluginSourceForCDN({
   url,
   source,
   transformSourceMapURL = false,
+  transformAssets = true,
 }: {
   url: string;
   source: string;
   transformSourceMapURL?: boolean;
+  transformAssets?: boolean;
```
Comment (leventebalogh) [inline public/app/features/plugins/cdn/utils.ts:15]: Nit: I guess it doesn't need to be an optional property if it has a default value?

File: public/app/features/plugins/sandbox/code_loader.ts [hunk 2]
```diff
@@ -19,8 +19,13 @@ export async function loadScriptIntoSandbox(url: string, meta: PluginMeta, sandb
...
     scriptCode = await response.text();
-    scriptCode = patchPluginSourceMap(meta, scriptCode);
-
+    //even though this is not loaded via a CDN we need to transform the sourceMapUrl
+    scriptCode = transformPluginSourceForCDN({
+      url,
+      source: scriptCode,
+      transformSourceMapURL: true,
+      transformAssets: false,
+    });
```
Comment (leventebalogh) [inline public/app/features/plugins/sandbox/code_loader.ts:28]: Do we also need to do it in `getPluginCode()` [here](https://github.com/grafana/grafana/blob/677a2ad4d8a634488508c22102dd8fb3096a95c9/public/app/features/plugins/sandbox/code_loader.ts#L60)?

**Review by leventebalogh — APPROVED** [review]
Looks good to me, left two questions.

File: public/app/features/plugins/cdn/utils.ts (hunk 1 above)
Comment (academo) [reply public/app/features/plugins/cdn/utils.ts:15]: If you don't make it optional it'll fail other places where you call it and not directly pass the parameter.

Comment (leventebalogh) [reply public/app/features/plugins/cdn/utils.ts:15]: Ahh, true, it's not like a function parameter with a default value.

File: public/app/features/plugins/sandbox/code_loader.ts (hunk 2 above)
Comment (academo) [reply public/app/features/plugins/sandbox/code_loader.ts:28]: Good idea. I'll replace that code with this one

File: public/app/features/plugins/sandbox/code_loader.ts [hunk 3]
```diff
@@ -57,7 +65,12 @@ export async function getPluginCode(meta: PluginMeta): Promise<string> {
     const pluginCodeUrl = resolveWithCache(meta.module);
     const response = await fetch(pluginCodeUrl);
     let pluginCode = await response.text();
-    pluginCode = patchPluginSourceMap(meta, pluginCode);
```
Comment (academo) [inline public/app/features/plugins/sandbox/code_loader.ts:60]: I'm replacing this function with the CDN one since it works better

**Review by leventebalogh — APPROVED** [review]

**Pull request merged**